### PR TITLE
Request compressed metadata json

### PR DESCRIPTION
**Pull request by zachthompson**
Request compressed metadata json from duck.co endpoint.   Still handles text/html responses.  Will need to be test on a server where json is being gzipped (ping @jkv).

### CODE REVIEW

**Comment by jkv** [comment]
@moollaza Let me know when you want to test this.

File: lib/DDG/Meta/Data.pm [hunk 1]
```diff
@@ -38,14 +38,16 @@ unless(%ia_metadata){
...
-        my $res = $ua->mirror("https://duck.co/ia/repo/all/json", $f);
+        $ua->default_header('Accept-Encoding' => scalar HTTP::Message::decodable());
+        my $res = $ua->mirror('https://duck.co/ia/repo/all/json', $f);
         unless($res->is_success || $res->code == 304){
             debug && warn "Failed to download metdata: " . $res->status_line;
         }
     }
 
-    my $metadata = decode_json(io($f)->all);
+    open my $fh, -B $f ? '<:gzip' : '<', $f or die "Failed to open file $f: $!";
```
Comment (nilnilnil) [inline lib/DDG/Meta/Data.pm]: I don't think this readable enough.

Comment (nilnilnil) [inline lib/DDG/Meta/Data.pm]: The whole block, not just this line.

Comment (zachthompson) [inline lib/DDG/Meta/Data.pm]: Give me some specifics and what you mean by "readable" so I can help clarify.

Comment (nilnilnil) [inline lib/DDG/Meta/Data.pm]: The -B confused me in the ternary at first. Then the `do { local $/; <$fh> }` block below.

Comment (zachthompson) [inline lib/DDG/Meta/Data.pm]: :+1: Let me see what I can do.

**Comment by nilnilnil** [comment]
Thanks @zachthompson. Also I'm @nilnilnil here. Someone took the other one! :smile:

From my perspective, this looks like we can merge now because we support both file reading methods.

Is there a formal test that is in process? I see questions about it; but someone needs to take the reigns, or merge and test.

:+1:

**Comment by zachthompson** [comment]
I've tested the gzip part on @jkv's instance.  He can probably bring it up if you want to do the same.

**Pull request merged**